### ISSUE TITLE
Blocking on a greenlet that does its own runtime imports during import time leads to LoopExit

### ISSUE THREAD

**Issue by jamadden** [gevent/gevent#728]
See benoitc/gunicorn#1192

**Comment by jamadden** [comment]
I'm not sure this should be considered a blocker, because if you do the same thing with threads instead, the process simply hangs forever, deadlocked on the native import lock. It would be nice if gevent could do better than that, but, because the native solution also fails, and gevent at least fails with an error message instead of silently hanging, we may be in pretty good shape.

```
$ cat test.py
import threading

def fn():
	import netrc

t = threading.Thread(target=fn)
t.start()
t.join()

$ python -c 'import test'
... # wait and wait and wait
^\Quit: 3
```

(In fact, in this simple example, if you monkey-patch everything at the top of the file, the gevent version does finish. It's only in the more complicated gunicorn example that gevent fails; I haven't boiled that down to a simpler test case yet.)

OTOH, Python 3 doesn't have this problem anymore and it might be nice to "backport" that functionality. That would require a better understanding of the arguments to the `__import__` function and implementing a per-module lock system.

**Comment by jamadden** [comment]
Then again, on the gripping hand, gevent 1.0 also didn't have this problem, so this is technically a regression.

**Comment by jamadden** [comment]
Fixed with d96b2a2
